proposals/b:zoe1/performActions.ts

```diff
@@ -0,0 +1,287 @@
+#!/usr/bin/env tsx
+
+import { execFileSync } from 'child_process'; // TODO: use execa
+import * as fsp from 'fs/promises';
+import * as path from 'path';
+import * as process from 'process';
+
+import { ZipReader } from '@endo/zip';
+import dbOpen from 'better-sqlite3';
+
+import assert from 'assert';
+import { makeAgd } from '../../upgrade-test-scripts/lib/agd-lib.js';
+import { agoric } from '../../upgrade-test-scripts/lib/cliHelper.js';
+import { voteLatestProposalAndWait } from '../../upgrade-test-scripts/lib/commonUpgradeHelpers.js';
+import { dbTool } from '../../upgrade-test-scripts/lib/vat-status.js';
+import { type WebCache } from '../../upgrade-test-scripts/lib/webAsset.js';
+import {
+  type ProposalInfo,
+  bundleDetail,
+  ensureISTForInstall,
+  flags,
+  getContractInfo,
+  loadedBundleIds,
+  txAbbr,
+} from './core-eval-support.js';
+
+// TODO move into core-eval-support
+const readSubmissions = async () => {
+  const files = await fsp.readdir('submission');
```

Comment:
> so that when I see calls to readSubmissions, I know there's an extra implicit argument
Why do you need to know that?

Reply:
because normally, return values of functions only depend on the args. In particular, a 0-argument function can only return a constant (again, modulo allocation).

This is normal local reasoning. The alternative is a burden to know the implementation details of every function in the system.